app/controllers/password_resets_controller.rb

```diff
@@ -0,0 +1,65 @@
+class PasswordResetsController < ApplicationController
+  before_action :find_user, only: :create
+  before_action :get_user,   only: [:edit, :update]
+  before_action :valid_user, only: [:edit, :update]
+  before_action :check_expiration, only: [:edit, :update]
+
+  def new; end
+
+  def edit; end
+
+  def create
+    if @user
+      @user.create_reset_digest
+      @user.send_password_reset_email
+      flash[:info] = t "password_resets.create.email_send_with_password_reset"
+      redirect_to root_url
+    else
+      flash.now[:danger] = t "password_resets.create.email_address_not_found"
+      render :new
+    end
+  end
+
+  def update
+    if params[:user][:password].empty?
```

Comment:
giải thích hoạt động của action này

Reply:
Cái này khi người dùng nhập form bao gồm pass mới và nhập lại pass đó, ta phải thêm kiểm tra rằng pass nhập vào có trống hay không. Nếu cái pass đó trống thì nó sẽ trở về cái pass cũ và quá trình đổi pass của chúng ta trở nên vô nghĩa. Còn nếu thành công(pass đúng, trùng nhau,...) thì log in vào user đó và thông báo rằng quá trình đổi pass đã thành công